Run syncer in kubernetes default namespace.

briantkennedy opened the pull request:
* Add build rules for syncer docker image
* Add yaml for starting syncer in a pod
* Add script for creating role account

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mdruskin/kubernetes-enterprise-control/43)
<!-- Reviewable:end -->

mdruskin commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 10 files reviewed at latest revision, 2 unresolved discussions.

---

*[deploy/syncer/deploy-gke.sh, line 64 at r1](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/43#-Kv8OqqMmrJYiLF6pe7C:-Kv8OqqMmrJYiLF6pe7D:bqjuaa7) ([raw file](https://github.com/mdruskin/kubernetes-enterprise-control/blob/b0e85131920a48068c10065937268d4e5792fcbe/deploy/syncer/deploy-gke.sh#L64)):*
> us.gcr.io/${PROJECT_ID}/${IMAGE}
--
${GCP_IMAGE}

---

*[deploy/syncer/deploy-gke.sh, line 69 at r1](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/43#-Kv8Oz7NBmD244d7mbj8:-Kv8Oz7NBmD244d7mbj9:brfe8su) ([raw file](https://github.com/mdruskin/kubernetes-enterprise-control/blob/b0e85131920a48068c10065937268d4e5792fcbe/deploy/syncer/deploy-gke.sh#L69)):*
> ```Shell
> 
> m4 -DIMAGE_NAME=${GCP_IMAGE} < tpl/syncer-pod.yaml > syncer-pod-gke.yaml
> kubectl create -f syncer-pod-gke.yaml
> ```

I think you should delete/create here. I'd imagine you want to run this script when you make a change.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/43#-:-Kv8R6Ac3UzyEqsyAWeC:bnfp4nl)*
<!-- Sent from Reviewable.io -->

briantkennedy commented:
Review status: 0 of 11 files reviewed at latest revision, 2 unresolved discussions.

---

*[deploy/syncer/deploy-gke.sh, line 64 at r1](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/43#-Kv8OqqMmrJYiLF6pe7C:-Kv8_zd9qFg-994J58Yr:b2xgd8a) ([raw file](https://github.com/mdruskin/kubernetes-enterprise-control/blob/b0e85131920a48068c10065937268d4e5792fcbe/deploy/syncer/deploy-gke.sh#L64)):*
<details><summary><i>Previously, mdruskin (Michael Druskin) wrote…</i></summary><blockquote>

> us.gcr.io/${PROJECT_ID}/${IMAGE}
--
${GCP_IMAGE}
</blockquote></details>

oh, heh, forgot that one, thanks

---

*[deploy/syncer/deploy-gke.sh, line 69 at r1](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/43#-Kv8Oz7NBmD244d7mbj8:-Kv8gMA9lYBBNoHm22Un:b-vmp9rs) ([raw file](https://github.com/mdruskin/kubernetes-enterprise-control/blob/b0e85131920a48068c10065937268d4e5792fcbe/deploy/syncer/deploy-gke.sh#L69)):*
<details><summary><i>Previously, mdruskin (Michael Druskin) wrote…</i></summary><blockquote>

I think you should delete/create here. I'd imagine you want to run this script when you make a change.
</blockquote></details>

Fixed, also added common script to reduce redundancy

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mdruskin/kubernetes-enterprise-control/43)*
<!-- Sent from Reviewable.io -->
